src/Gaffer/ScriptNode.cpp

```diff
@@ -970,9 +968,18 @@ void ScriptNode::plugSet( Plug *plug )
 	{
 		const boost::filesystem::path fileName( fileNamePlug()->getValue() );
 		context()->set( g_scriptName, fileName.stem().string() );
+
+		bool isReadOnly = false;
+		if( boost::filesystem::exists( fileName ) )
+		{
+			std::ofstream testOpen( fileName.c_str(), std::fstream::app );
+			isReadOnly = !testOpen.is_open();
+			testOpen.close();
+		}
```

Comment:
I didn't know about `_access`, that's a much better solution. I went with the `#define` method since I also am declaring the `W_OK`. Happy to use `inline` if that's better. Fixed in 9367ec0436ff84538ccfa1262ad157eb7e726dd6